pkg/machine/qemu/machine_unix.go

```diff
@@ -12,6 +12,47 @@ import (
 	"golang.org/x/sys/unix"
 )
 
+// Performs environment setup for qemu child process. Returns function to undo
+// all changes made.
+func setupQemuEnv() (undoFn func(), err error) {
+	// store functions that will revert changes
+	undoFns := make([]func(), 0)
+
+	// iterates all the undo functions to revert changes made
+	undoFn = func() {
+		for idx := range undoFns {
+			undoFns[idx]()
+		}
+	}
+
+	// The following changes the max open file count, as seen in `ulimit -n`
+	// This is necessary to prevent the virtiofs feature in qemu from being
+	// limited when started from a user session, as observed on MacOS.
+
+	// 1) preserve current rLimit value for undo function
+	var rLimit syscall.Rlimit
+	err = syscall.Getrlimit(syscall.RLIMIT_NOFILE, &rLimit)
+	if err != nil {
+		return
+	}
+
+	// 2) add callback for rLimit to be reverted
+	undoFns = append(undoFns, func() {
+		syscall.Setrlimit(syscall.RLIMIT_NOFILE, &rLimit)
+	})
+
+	// 3) change rLimit to maximum value
+	err = syscall.Setrlimit(syscall.RLIMIT_NOFILE, &syscall.Rlimit{
+		Max: syscall.RLIM_INFINITY,
```

Comment:
Setting the limit this high can potentially fail. On Linux anything greater than  /proc/sys/fs/nr_open can throw EPERM, and on Mac OS, some versions can through EINVAL if you exceed OPEN_MAX(10240). There could also be cases where the hard limit is set lower than the system max, in which you could raise the soft to match the hard limit. Most likely podman machine is not running as root, so that could be the solution to just set soft to match the current hard limit

Reply:
Can't find solid documentation on it, so I'm kinda taking a guess that `syscall.Rlimit.Max` is the hard limit and `syscall.Rlimit.Cur` is the soft limit.

Does the last change I made resolve this?

Reply:
Yes, looks good ant thats right Cur = soft. FYI you can find the docs in the OSX/Linux man pages `man setrlimit`